esp-hal/src/parl_io.rs

```diff
@@ -1460,47 +1494,37 @@ fn internal_init(frequency: HertzU32) -> Result<(), Error> {
     Ok(())
 }
 
-impl<Dm> ParlIoTx<'_, Dm>
+impl<'d, Dm> ParlIoTx<'d, Dm>
 where
     Dm: DriverMode,
 {
     /// Perform a DMA write.
     ///
-    /// This will return a [DmaTransferTx]
+    /// This will return a [ParlIoTxTransfer]
     ///
     /// The maximum amount of data to be sent is 32736 bytes.
-    pub fn write_dma<'t, TXBUF>(
-        &'t mut self,
-        words: &'t TXBUF,
-    ) -> Result<DmaTransferTx<'t, Self>, Error>
+    pub fn write<BUF>(
+        mut self,
+        number_of_bytes: usize,
+        mut buffer: BUF,
+    ) -> Result<ParlIoTxTransfer<'d, BUF, Dm>, Error>
```

Comment:
On error this eats self and the BUF - it should return a tuple right?

Reply:
Ahh you're right. I'll update that